rust/processor/src/db/common/models/fungible_asset_models/raw_v2_coin_to_fungible_asset_mappings.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use super::raw_v2_fungible_metadata::RawFungibleAssetMetadataModel;
+use crate::{
+    db::common::models::fungible_asset_models::raw_v2_fungible_asset_balances::get_paired_metadata_address,
+    schema::coin_to_fungible_asset_mappings, utils::database::DbPoolConnection,
+};
+use ahash::AHashMap;
+use diesel::query_dsl::methods::SelectDsl;
+use diesel_async::RunQueryDsl;
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+
+pub type CoinToFungibleAssetMappings = AHashMap<String, String>;
+pub type CoinToFungibleAssetMappingsForDB = AHashMap<String, RawCoinToFungibleAssetMapping>;
+
+#[derive(Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(coin_type))]
+#[diesel(table_name = coin_to_fungible_asset_mappings)]
+pub struct RawCoinToFungibleAssetMapping {
+    pub coin_type: String,
+    pub fungible_asset_metadata_address: String,
+    pub last_transaction_version: i64,
+}
+
+impl RawCoinToFungibleAssetMapping {
+    pub fn from_raw_fungible_asset_metadata(metadata: &RawFungibleAssetMetadataModel) -> Self {
+        let fungible_asset_metadata_address = get_paired_metadata_address(&metadata.asset_type);
+        Self {
+            coin_type: metadata.asset_type.clone(),
+            fungible_asset_metadata_address,
+            last_transaction_version: metadata.last_transaction_version,
+        }
+    }
+
+    /// This should be triggered on startup only. After that we will use the mapping in memory.
+    pub async fn get_all_mappings(conn: &mut DbPoolConnection<'_>) -> CoinToFungibleAssetMappings {
+        match Self::query_all(conn).await {
+            Ok(mappings) => {
+                let mut result = AHashMap::new();
+                for (metadata_address, coin_type) in mappings {
+                    result.insert(coin_type, metadata_address);
+                }
+                result
+            },
+            Err(e) => {
+                tracing::error!(
+                    error = ?e,
+                    "Failed to query coin to fungible asset mappings"
+                );
+                panic!("Failed to query coin to fungible asset mappings: {:?}", e);
+            },
+        }
+    }
+
+    async fn query_all(
+        conn: &mut DbPoolConnection<'_>,
+    ) -> diesel::QueryResult<Vec<(String, String)>> {
+        coin_to_fungible_asset_mappings::table
+            .select((
+                coin_to_fungible_asset_mappings::fungible_asset_metadata_address,
+                coin_to_fungible_asset_mappings::coin_type,
```

Comment:
Lets flip the order of these so coin_type comes first cuz coin_type is the PK and so we wont need to flip them again later